src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilder.java

```diff
@@ -0,0 +1,447 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.sensitivityanalysis.server.service;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Report;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.TypedValue;
+import com.powsybl.contingency.Contingency;
+import com.powsybl.contingency.ContingencyContext;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.IdentifiableType;
+import com.powsybl.iidm.network.Load;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.sensitivity.SensitivityFactor;
+import com.powsybl.sensitivity.SensitivityFunctionType;
+import com.powsybl.sensitivity.SensitivityVariableSet;
+import com.powsybl.sensitivity.SensitivityVariableType;
+import com.powsybl.sensitivity.WeightedSensitivityVariable;
+import org.apache.commons.lang3.StringUtils;
+import org.gridsuite.sensitivityanalysis.server.dto.IdentifiableAttributes;
+import org.gridsuite.sensitivityanalysis.server.dto.SensitivityAnalysisInputData;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class SensitivityAnalysisInputBuilder {
+    private static final String EXPECTED_TYPE = "expectedType";
+
+    private final Network network;
+    private final SensitivityAnalysisRunContext context;
+    private final ActionsService actionsService;
+    private final FilterService filterService;
+
+    private List<Contingency> contingencies = new ArrayList<>();
+    private List<SensitivityFactor> factors = new ArrayList<>();
+    private List<SensitivityVariableSet> variablesSets = new ArrayList<>();
+
+    public SensitivityAnalysisInputBuilder(Network network, SensitivityAnalysisRunContext context, ActionsService actionsService, FilterService filterService) {
+        this.network = network;
+        this.context = context;
+        this.actionsService = actionsService;
+        this.filterService = filterService;
+    }
+
+    private List<Contingency> buildContingencies(List<SensitivityAnalysisInputData.Ident> contingencyListsIdents) {
+        return contingencyListsIdents.stream()
+            .flatMap(contingencyListIdent -> actionsService.getContingencyList(contingencyListIdent.getId(), context.getNetworkUuid(), context.getVariantId()).stream())
+            .collect(Collectors.toList());
+    }
+
+    private double getGeneratorWeight(Generator generator, SensitivityAnalysisInputData.DistributionType distributionType, Double distributionKey) {
+        double weight;
```

Comment:
You may use the switch return style here , that would save the forgettable `break`s ;-)

Reply:
Done